natter-api/src/main/java/com/manning/apisecurityinaction/controller/SpaceController.java

```diff
@@ -50,85 +56,120 @@ public JSONObject createSpace(Request request, Response response) {
   public JSONObject postMessage(Request request, Response response) {
```

Comment:
Code review for postMessage function:

1. Authorization check is missing
2. No check for null or empty string for both user & message that may cause crash
3. No check for input type for both user & message
4. Adding inputs (msg) to query with no validation may cause sql injection issues.
5. spaceId should not be added in url and it can be added if it is encrypted
6. There is no check that the spaceId is owned by logged in user, so hacker can post message to any space
7. No error handling for missing input/param
8. No error handling for forbidden status
9. No unit testing for above function
10. Description of function and parameters is not added
11. Remove empty lines